### PR TITLE
Implement shift as part of IO

### LINKED ISSUE #105: Implement `shift` combinator
I think one of the useful features that we need to add to ZIO is a thread shifting. 
Like `io.executeOn(ioScheduler)` in Monix

### PR DESCRIPTION
**Pull request by mijicd**
Closes #105

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 target
 .sbtopts
+.idea/
```
Comment (hrhino) [inline .gitignore]: please read lines 5-6

File: core/shared/src/main/scala/scalaz/zio/IO.scala [hunk 2]
```diff
@@ -788,6 +790,13 @@ object IO {
       )
     )
 
+  final def shift[E, A](ec: ExecutionContext): IO[Nothing, Unit] =
+    IO.async { cb: Callback[Nothing, Unit] =>
+      ec.execute(new Runnable {
```
Comment (regiskuckaertz) [inline core/shared/src/main/scala/scalaz/zio/IO.scala]: This belongs in `RTS` where an `ExecutionContext` will be provided.

File: .gitignore (hunk 1 above)
Comment (mijicd) [reply .gitignore]: @hrhino thanks, my bad, removed.

File: core/shared/src/main/scala/scalaz/zio/IO.scala (hunk 2 above)
Comment (jmcardon) [reply core/shared/src/main/scala/scalaz/zio/IO.scala]: Not that it's a big deal, but are you essentially indicating that `zio` does not plan to support pool shifting outside of the one defined in the `RTS`?

**Comment by jdegoes** [comment]
@mielientiev Looks good. Thank you, & congrats on your first contribution.

File: core/shared/src/main/scala/scalaz/zio/IO.scala (hunk 2 above)
Comment (regiskuckaertz) [reply core/shared/src/main/scala/scalaz/zio/IO.scala]: don't know the plans save for [this](https://github.com/scalaz/scalaz-zio/issues/36), I'm just a happy contributor 🤗

Comment (mijicd) [reply core/shared/src/main/scala/scalaz/zio/IO.scala]: As far as I understood @jdegoes, this is just a stepping stone to the #36, so I think we're all on the same page :).

**Pull request merged**